wallets/core/src/namespaces/evm/actions.ts

```diff
@@ -0,0 +1,104 @@
+import type { EvmActions, ProviderApi } from './types.js';
+import type { Context, Subscriber } from '../../hub/namespaces/mod.js';
+import type { SubscriberCleanUp } from '../../hub/namespaces/types.js';
+import type { CaipAccount, FunctionWithContext } from '../common/types.js';
+import type { EIP1193EventMap } from 'viem';
+
+import { AccountId } from 'caip';
+
+import { recommended as commonRecommended } from '../common/actions.js';
+
+import { CAIP_NAMESPACE } from './constants.js';
+import { getAccounts, switchOrAddNetwork } from './utils.js';
+
+export const recommended = [...commonRecommended];
+
+export function connect(
+  instance: () => ProviderApi
+): ['connect', FunctionWithContext<EvmActions['connect'], Context>] {
+  return [
+    'connect',
+    async (_context, chain) => {
+      const evmInstance = instance();
```

Comment:
Can you elaborate how eager connect is implemented in EVM or Solana? I wasn't able to find related code.

Reply:
Please take a look at `wallets/react/src/hub/useAutoConnect.ts` for hub, and `wallets/react/src/legacy/useAutoConnect.ts` for legacy.